src/render/decorations/IHyprWindowDecoration.cpp

```diff
@@ -2,12 +2,37 @@
 
 #include "../../Window.hpp"
 
+IHyprWindowDecoration::IHyprWindowDecoration(CWindow* pWindow) {
+    m_pWindow = pWindow;
+}
+
 IHyprWindowDecoration::~IHyprWindowDecoration() {}
 
 SWindowDecorationExtents IHyprWindowDecoration::getWindowDecorationReservedArea() {
     return SWindowDecorationExtents{};
 }
 
+wlr_box IHyprWindowDecoration::getWindowDecorationBox() {
```

Comment:
we could use a CRegion here. Would also be easier to check against later.

Probably can make a simple CRegion::pointInside(const Vector2D&) as a helper

solves also the issue with the box of the window not being a part of the deco

Reply:
done